runtime/lib/backend/LightningKokkosSimulator.cpp

```diff
@@ -46,15 +46,15 @@ auto LightningKokkosSimulator::GetNumQubits() const -> size_t
 
 void LightningKokkosSimulator::StartTapeRecording()
 {
-    RT_FAIL_IF(this->cache_recording, "Cannot re-activate the cache manager");
-    this->cache_recording = true;
+    RT_FAIL_IF(this->tape_recording, "Cannot re-activate the cache manager");
+    this->tape_recording = true;
     this->cache_manager.Reset();
 }
 
 void LightningKokkosSimulator::StopTapeRecording()
 {
-    if (this->cache_recording) {
-        this->cache_recording = false;
+    if (this->tape_recording) {
+        this->tape_recording = false;
     }
```

Comment:
Because, it doesn't make sense to re-activate the cache manager and if this happens it's highly related to an incorrect use of this manager. There is also no harm with stopping the manager and as such no error associated with this.

Reply:
Hmm I don't see the difference, it doesn't technically harm execution on either stopping or starting twice, but like you say it's probably indicative of a program logic error. Both should be treated the same imo.